docs/migration/migration7.md

```diff
@@ -0,0 +1,244 @@
+    {
+    }
+
+    // overridden methods here
+}
+
+app.UseGraphQL<MyMiddleware>(new GraphQLHttpMiddlewareOptions());
+```
+
+</p></details>
+
+<details><summary>With separate subscription endpoint</summary><p>
+
+```csharp
+// v6
+app.UseGraphQL<MySchema>("/graphql");
+app.UseGraphQLWebSockets<MySchema>("/graphqlsubscription");
+
+// v7
+app.UseGraphQL<MySchema>("/graphql", o => o.HandleWebSockets = false);
+app.UseGraphQL<MySchema>("/graphqlsubscription", o => {
+    o.HandleGet = false;
+    o.HandlePost = false;
+});
+```
+
+</p></details>
+
+<details><summary>To retain prior media type of `application/json`</summary><p>
+
+```csharp
+class MyMiddleware<TSchema> : GraphQLHttpMiddleware<TSchema>
+    where TSchema : ISchema
+{
+    public MyMiddleware(
+        RequestDelegate next,
+        IGraphQLTextSerializer serializer,
+        IDocumentExecuter<TSchema> documentExecuter,
+        IServiceScopeFactory serviceScopeFactory,
+        GraphQLHttpMiddlewareOptions options,
+        IHostApplicationLifetime hostApplicationLifetime)
+        : base(next, serializer, documentExecuter, serviceScopeFactory, options, hostApplicationLifetime)
+    {
+    }
+
+    protected override string SelectResponseContentType(HttpContext context)
+        => "application/json";
+}
+
+app.UseGraphQL<MyMiddleware<ISchema>>("/graphql", new GraphQLHttpMiddlewareOptions());
+```
+
+</p></details>
+
+<details><summary>If you had code within the `RequestExecutedAsync` protected method</summary><p>
+
+Either override `HandleRequestAsync`, `HandleBatchRequestAsync` and/or `ExecuteRequestAsync`,
+or call the builder method `ConfigureExecution` to add code before/after the call to `IDocumentExecuter.ExecuteAsync`.
+
+</p></details>
+
+## Migration of user context builder
+
+If you used the `AddUserContextBuilder` builder method, there are no changes necessary.
+For custom implementations of `IUserContextBuilder`, you will need to update the method
+signature for `BuildUserContextAsync`.
+
+## Migration of authorization validation rule
+
+| :warning: **Note that authorization rules on input types are ignored in v7** :warning: |
+| --- |
+
+If you need `IClaimsPrincipalAccessor`,  `IAuthorizationErrorMessageBuilder`, or the detailed authorization
+failure messages provided in v6, then you may use the deprecated authorization rule with no code changes.
+Please open an issue within GitHub explaining your need so it may be addressed.
+
+Otherwise, remove the GraphQL.Server.Authorization.AspNetCore NuGet package and make changes as shown below:
+
+```csharp
+// v6
+services.AddGraphQL(b => b
+    .AddGraphQLAuthorization(options => {
+        // ASP.NET authorization configuration
+    })
+    // other code
+);
+
+// v7
+services.AddGraphQL(b => b
+    .AddAuthorizationRule()
+    // other code
+);
+services.AddAuthorization(options => {
+    // ASP.NET authorization configuration
+});
+```
+
+## Migration of UI middleware
+
+No changes are necessary for UI middleware
```

Comment:
#862

Reply:
```suggestion
The path must now be specified prior to the options class, rather than after.

```csharp
// v6/v7
app.UseGraphQLPlayground();

// v6/v7
app.UseGraphQLPlayground("/");

// v6
app.UseGraphQLPlayground(new PlaygroundOptions(), "/");

// v7
app.UseGraphQLPlayground("/", new PlaygroundOptions());

// v6
app.UseGraphQLPlayground(new PlaygroundOptions());

// v7
app.UseGraphQLPlayground(options: new PlaygroundOptions());
```

Reply:
@sungam3r does this look okay?

Reply:
I can't add a suggestion with the trailing triple-backslash, so I'll add that manually.

Reply:
ok